src/main/java/Rectangle.java

```diff
@@ -1,63 +1,94 @@
+import java.util.Objects;
+
 class Rectangle {
 
-    private Point pos;
-    private Point size;
+    private Point leftTop;
+    private Point bottomRight;
+
+    Rectangle(){
+        this.leftTop = new Point(0, 0);
+        this.bottomRight = new Point(0, 0);
+    }
 
-    Rectangle(Point pos, Point size){
-        this.pos = pos;
-        this.size = size;
+    Rectangle(Point leftTop, Point bottomRight){
+        this.leftTop = leftTop;
+        this.bottomRight = bottomRight;
     }
 
-    public Point getPos() {
-        return pos;
+    public Point getLeftTop() {
+        return leftTop;
     }
 
-    public Point getSize() {
-        return size;
+    public Point getBottomRight() {
+        return bottomRight;
     }
 
     public boolean intersects(Rectangle rectangle){
-        if (rectangle.pos.x > pos.x + size.x){
+        if (rectangle.leftTop.x > bottomRight.x){
             return false;
         }
 
-        if (rectangle.pos.y > pos.y + size.y){
+        if (rectangle.leftTop.y > bottomRight.y){
             return false;
         }
 
-        if (rectangle.pos.x + rectangle.size.x < pos.x){
+        if (rectangle.bottomRight.x < leftTop.x){
             return false;
         }
 
-        if (rectangle.pos.y + rectangle.size.y < pos.y){
+        if (rectangle.bottomRight.y < leftTop.y){
             return false;
         }
 
         return true;
     }
 
     public boolean includes(Rectangle rectangle){
-        if (rectangle.getPos().x < pos.x){
+        if (rectangle.getLeftTop().x < leftTop.x){
             return false;
         }
 
-        if (rectangle.getPos().y < pos.y){
+        if (rectangle.getLeftTop().y < leftTop.y){
             return false;
         }
 
-        if (rectangle.getPos().x + rectangle.getSize().x > pos.x + size.x){
+        if (rectangle.getBottomRight().x > bottomRight.x){
             return false;
         }
 
-        if (rectangle.getPos().y + rectangle.getSize().y > pos.y + size.y){
+        if (rectangle.getBottomRight().y > bottomRight.y){
             return false;
         }
 
         return true;
     }
 
+    public void setRectangle(Rectangle rectangle){
+        this.leftTop.x = rectangle.getLeftTop().x;
+        this.leftTop.y = rectangle.getLeftTop().y;
+        this.bottomRight.x = rectangle.getBottomRight().x;
+        this.bottomRight.y = rectangle.getBottomRight().y;
+    }
+
     public void shift(Point shift){
-        this.pos.x = this.pos.x + shift.x;
-        this.pos.y = this.pos.y + shift.y;
+        this.leftTop.x = this.leftTop.x + shift.x;
```

Comment:
1) this is used only for field initialization
2) use +=

```
leftTop.x += shift.x;
```

I would add new Point method:
point.shift();